File: split/endfilereading.pas
procedure endfilereading;
begin
  First := curinput.startfield;
  line := linestack[curinput.indexfield];
  if curinput.namefield > 17 then
    aclose(inputfile[curinput.indexfield]);
  begin
    inputptr := inputptr - 1;
    curinput := inputstack[inputptr];
  end;
  inopen := inopen - 1;
end;

File: cpp/endfilereading.cpp
#include "endfilereading.h"
#include "aclose.h"

void endfilereading(void)
{
	First = curinput.startfield;
	line = linestack[curinput.indexfield];
	if (curinput.namefield > 17)
		aclose(inputfile[curinput.indexfield]);
	inputptr--;
	curinput = inputstack[inputptr];
	inopen--;
}
